forest/shared/src/cli/mod.rs

```diff
@@ -204,20 +280,44 @@ impl ConfigPath {
     }
 }
 
-fn find_config_path(opts: &CliOpts) -> Option<ConfigPath> {
-    if let Some(s) = &opts.config {
-        return Some(ConfigPath::Cli(PathBuf::from(s)));
-    }
-    if let Ok(s) = std::env::var("FOREST_CONFIG_PATH") {
-        return Some(ConfigPath::Env(PathBuf::from(s)));
-    }
-    if let Some(dir) = ProjectDirs::from("com", "ChainSafe", "Forest") {
-        let path = dir.config_dir().join("config.toml");
-        if path.exists() {
-            return Some(ConfigPath::Project(path));
+pub enum Opts<'a> {
+    CliOpts(&'a CliOpts),
+    DaemonOpts(&'a DaemonOpts),
+}
+
+fn find_config_path(opts: Opts) -> Option<ConfigPath> {
+    match opts {
+        Opts::CliOpts(cli_opts) => {
+            if let Some(s) = &cli_opts.config {
+                return Some(ConfigPath::Cli(PathBuf::from(s)));
+            }
+            if let Ok(s) = std::env::var("FOREST_CONFIG_PATH") {
+                return Some(ConfigPath::Env(PathBuf::from(s)));
+            }
+            if let Some(dir) = ProjectDirs::from("com", "ChainSafe", "Forest") {
+                let path = dir.config_dir().join("config.toml");
+                if path.exists() {
+                    return Some(ConfigPath::Project(path));
+                }
+            }
+            None
+        }
+        Opts::DaemonOpts(daemon_opts) => {
+            if let Some(s) = &daemon_opts.config {
+                return Some(ConfigPath::Cli(PathBuf::from(s)));
+            }
+            if let Ok(s) = std::env::var("FOREST_CONFIG_PATH") {
+                return Some(ConfigPath::Env(PathBuf::from(s)));
+            }
+            if let Some(dir) = ProjectDirs::from("com", "ChainSafe", "Forest") {
+                let path = dir.config_dir().join("config.toml");
+                if path.exists() {
+                    return Some(ConfigPath::Project(path));
+                }
+            }
+            None
```

Comment:
You've still duplicated the logic. The logic is exactly the same in both instances.